### PR TITLE
common/config: move internal UNIX socket path option to config file

### PR DESCRIPTION
**Pull request by ptrus**
TODO: backport

### CODE REVIEW

**Review by kostko — APPROVED** [review]

File: go/oasis-node/cmd/common/grpc/grpc.go [hunk 1]
```diff
@@ -75,15 +68,10 @@ func NewServerLocal(installWrapper bool) (*cmnGrpc.Server, error) {
 	if dataDir == "" {
```
Comment (peternose) [inline go/oasis-node/cmd/common/grpc/grpc.go:68]: Can this also be removed?

Comment (ptrus) [reply go/oasis-node/cmd/common/grpc/grpc.go:68]: Yeah I'll look into removing this in a follow up as this was already merged. I don't see a reason why the datadir was required here specifically - even before the change in this PR.

Comment (peternose) [reply go/oasis-node/cmd/common/grpc/grpc.go:68]: > I don't see a reason why the datadir was required here specifically - even before the change in this PR.

To construct the path, I guess.

Comment (ptrus) [reply go/oasis-node/cmd/common/grpc/grpc.go:68]: Yeah but this didn't change with the PR. It still uses the same data dir config value to construct it (unless overriden).

**Pull request merged**